src/app/app.component.ts

```diff
@@ -261,14 +321,80 @@ export class AppComponent implements AfterViewInit {
     this.entrySelected = treeNodesSelected?.length > 0 ? treeNodesSelected[0] : undefined;
   }
 
+  async onToolbarOptionSelected(event) {
+    const customEvent = event as CustomEvent<ToolbarOption>;
+    const optionSelected: ToolbarOption = customEvent.detail;
+    if (optionSelected.name == 'Refresh') {
+      await this.lfRepositoryBrowser?.nativeElement.refreshAsync();
+      console.log('refresh');
```

Comment:
these could be a "handler" in the toolbarOption so you just have to call that and it's already defined for you. I would prefer that since the names have spaces, etc.